Build failure: llvmPackages_git.lldb

### Steps To Reproduce

`nix-build -A llvmPackages_git.lldb` at 1cb26fc22c0a460d84c9582e882ab33d45b6a666

### Build log

```
[989/996] Linking CXX shared library lib/liblldb.so.18.0.0git
FAILED: lib/liblldb.so.18.0.0git
: && /nix/store/sfgnb6rr428bssyrs54d6d0vv2avi95c-gcc-wrapper-12.3.0/bin/g++ -fPIC -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -fno-lifetime-dse -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wno-missing-field-initializers -Wimplicit-fallthrough -Wno-nonnull -Wno-class-memaccess -Wno-redundant-move -Wno-pessimizing-move -Wno-noexcept-type -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wno-misleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -Wno-deprecated-declarations -Wno-unknown-pragmas -Wno-strict-aliasing -Wno-stringop-truncation -O3 -DNDEBUG  -Wl,-z,defs -Wl,-z,nodelete   -Wl,-rpath-link,/build/lldb-src-18.0.0-unstable-2023-10-04/lldb/build/./lib  -Wl,--gc-sections  -Wl,--version-script,"/build/lldb-src-18.0.0-unstable-2023-10-04/lldb/build/source/API/liblldb.exports" -shared -Wl,-soname,liblldb.so.18git -o lib/liblldb.so.18.0.0git source/API/CMakeFiles/liblldb.dir/SBAddress.cpp.o source/API/CMakeFiles/liblldb.dir/SBAttachInfo.cpp.o source/API/CMakeFiles/liblldb.dir/SBBlock.cpp.o source/API/CMakeFiles/liblldb.dir/SBBreakpoint.cpp.o source/API/CMakeFiles/liblldb.dir/SBBreakpointLocation.cpp.o source/API/CMakeFiles/liblldb.dir/SBBreakpointName.cpp.o source/API/CMakeFiles/liblldb.dir/SBBreakpointOptionCommon.cpp.o source/API/CMakeFiles/liblldb.dir/SBBroadcaster.cpp.o source/API/CMakeFiles/liblldb.dir/SBCommandInterpreter.cpp.o source/API/CMakeFiles/liblldb.dir/SBCommandInterpreterRunOptions.cpp.o source/API/CMakeFiles/liblldb.dir/SBCommandReturnObject.cpp.o source/API/CMakeFiles/liblldb.dir/SBCommunication.cpp.o source/API/CMakeFiles/liblldb.dir/SBCompileUnit.cpp.o source/API/CMakeFiles/liblldb.dir/SBData.cpp.o source/API/CMakeFiles/liblldb.dir/SBDebugger.cpp.o source/API/CMakeFiles/liblldb.dir/SBDeclaration.cpp.o source/API/CMakeFiles/liblldb.dir/SBEnvironment.cpp.o source/API/CMakeFiles/liblldb.dir/SBError.cpp.o source/API/CMakeFiles/liblldb.dir/SBEvent.cpp.o source/API/CMakeFiles/liblldb.dir/SBExecutionContext.cpp.o source/API/CMakeFiles/liblldb.dir/SBExpressionOptions.cpp.o source/API/CMakeFiles/liblldb.dir/SBFileSpec.cpp.o source/API/CMakeFiles/liblldb.dir/SBFile.cpp.o source/API/CMakeFiles/liblldb.dir/SBFileSpecList.cpp.o source/API/CMakeFiles/liblldb.dir/SBFrame.cpp.o source/API/CMakeFiles/liblldb.dir/SBFunction.cpp.o source/API/CMakeFiles/liblldb.dir/SBHostOS.cpp.o source/API/CMakeFiles/liblldb.dir/SBInstruction.cpp.o source/API/CMakeFiles/liblldb.dir/SBInstructionList.cpp.o source/API/CMakeFiles/liblldb.dir/SBLanguageRuntime.cpp.o source/API/CMakeFiles/liblldb.dir/SBLaunchInfo.cpp.o source/API/CMakeFiles/liblldb.dir/SBLineEntry.cpp.o source/API/CMakeFiles/liblldb.dir/SBListener.cpp.o source/API/CMakeFiles/liblldb.dir/SBMemoryRegionInfo.cpp.o source/API/CMakeFiles/liblldb.dir/SBMemoryRegionInfoList.cpp.o source/API/CMakeFiles/liblldb.dir/SBModule.cpp.o source/API/CMakeFiles/liblldb.dir/SBModuleSpec.cpp.o source/API/CMakeFiles/liblldb.dir/SBPlatform.cpp.o source/API/CMakeFiles/liblldb.dir/SBProcess.cpp.o source/API/CMakeFiles/liblldb.dir/SBProcessInfo.cpp.o source/API/CMakeFiles/liblldb.dir/SBQueue.cpp.o source/API/CMakeFiles/liblldb.dir/SBQueueItem.cpp.o source/API/CMakeFiles/liblldb.dir/SBReproducer.cpp.o source/API/CMakeFiles/liblldb.dir/SBScriptObject.cpp.o source/API/CMakeFiles/liblldb.dir/SBSection.cpp.o source/API/CMakeFiles/liblldb.dir/SBSourceManager.cpp.o source/API/CMakeFiles/liblldb.dir/SBStream.cpp.o source/API/CMakeFiles/liblldb.dir/SBStringList.cpp.o source/API/CMakeFiles/liblldb.dir/SBStructuredData.cpp.o source/API/CMakeFiles/liblldb.dir/SBSymbol.cpp.o source/API/CMakeFiles/liblldb.dir/SBSymbolContext.cpp.o source/API/CMakeFiles/liblldb.dir/SBSymbolContextList.cpp.o source/API/CMakeFiles/liblldb.dir/SBTarget.cpp.o source/API/CMakeFiles/liblldb.dir/SBThread.cpp.o source/API/CMakeFiles/liblldb.dir/SBThreadCollection.cpp.o source/API/CMakeFiles/liblldb.dir/SBThreadPlan.cpp.o source/API/CMakeFiles/liblldb.dir/SBTrace.cpp.o source/API/CMakeFiles/liblldb.dir/SBTraceCursor.cpp.o source/API/CMakeFiles/liblldb.dir/SBType.cpp.o source/API/CMakeFiles/liblldb.dir/SBTypeCategory.cpp.o source/API/CMakeFiles/liblldb.dir/SBTypeEnumMember.cpp.o source/API/CMakeFiles/liblldb.dir/SBTypeFilter.cpp.o source/API/CMakeFiles/liblldb.dir/SBTypeFormat.cpp.o source/API/CMakeFiles/liblldb.dir/SBTypeNameSpecifier.cpp.o source/API/CMakeFiles/liblldb.dir/SBTypeSummary.cpp.o source/API/CMakeFiles/liblldb.dir/SBTypeSynthetic.cpp.o source/API/CMakeFiles/liblldb.dir/SBValue.cpp.o source/API/CMakeFiles/liblldb.dir/SBValueList.cpp.o source/API/CMakeFiles/liblldb.dir/SBVariablesOptions.cpp.o source/API/CMakeFiles/liblldb.dir/SBWatchpoint.cpp.o source/API/CMakeFiles/liblldb.dir/SBWatchpointOptions.cpp.o source/API/CMakeFiles/liblldb.dir/SBUnixSignals.cpp.o source/API/CMakeFiles/liblldb.dir/SystemInitializerFull.cpp.o  -Wl,-rpath,"\$ORIGIN/..//nix/store/wklkp9jr0crsi54vdyj44ipaxb4q946p-lldb-18.0.0-unstable-2023-10-04-lib/lib:/nix/store/y6hbc39dib6casgmv1lz10s9mia2jqv9-llvm-18.0.0-unstable-2023-10-04-lib/lib:/nix/store/vbdakqf8mwamgm2s5jfbc54mf2mhljfw-clang-18.0.0-unstable-2023-10-04-lib/lib:"  lib/liblldbBreakpoint.a  lib/liblldbCore.a  lib/liblldbDataFormatters.a  lib/liblldbExpression.a  lib/liblldbHost.a  lib/liblldbInitialization.a  lib/liblldbInterpreter.a  lib/liblldbSymbol.a  lib/liblldbTarget.a  lib/liblldbUtility.a  lib/liblldbVersion.a  lib/liblldbPluginABIAArch64.a  lib/liblldbPluginABIARM.a  lib/liblldbPluginABIHexagon.a  lib/liblldbPluginABIMips.a  lib/liblldbPluginABIMSP430.a  lib/liblldbPluginABIPowerPC.a  lib/liblldbPluginABIRISCV.a  lib/liblldbPluginABISystemZ.a  lib/liblldbPluginABIX86.a  lib/liblldbPluginArchitectureArm.a  lib/liblldbPluginArchitectureMips.a  lib/liblldbPluginArchitecturePPC64.a  lib/liblldbPluginArchitectureAArch64.a  lib/liblldbPluginDisassemblerLLVMC.a  lib/liblldbPluginDynamicLoaderDarwinKernel.a  lib/liblldbPluginDynamicLoaderFreeBSDKernel.a  lib/liblldbPluginDynamicLoaderMacOSXDYLD.a  lib/liblldbPluginDynamicLoaderPosixDYLD.a  lib/liblldbPluginDynamicLoaderStatic.a  lib/liblldbPluginDynamicLoaderHexagonDYLD.a  lib/liblldbPluginDynamicLoaderWindowsDYLD.a  lib/liblldbPluginDynamicLoaderWasmDYLD.a  lib/liblldbPluginInstructionARM.a  lib/liblldbPluginInstructionARM64.a  lib/liblldbPluginInstructionLoongArch.a  lib/liblldbPluginInstructionMIPS.a  lib/liblldbPluginInstructionMIPS64.a  lib/liblldbPluginInstructionPPC64.a  lib/liblldbPluginInstructionRISCV.a  lib/liblldbPluginInstrumentationRuntimeASan.a  lib/liblldbPluginInstrumentationRuntimeMainThreadChecker.a  lib/liblldbPluginInstrumentationRuntimeTSan.a  lib/liblldbPluginInstrumentationRuntimeUBSan.a  lib/liblldbPluginJITLoaderGDB.a  lib/liblldbPluginCPlusPlusLanguage.a  lib/liblldbPluginObjCLanguage.a  lib/liblldbPluginObjCPlusPlusLanguage.a  lib/liblldbPluginCXXItaniumABI.a  lib/liblldbPluginAppleObjCRuntime.a  lib/liblldbPluginGNUstepObjCRuntime.a  lib/liblldbPluginMemoryHistoryASan.a  lib/liblldbPluginObjectContainerBSDArchive.a  lib/liblldbPluginObjectContainerMachOArchive.a  lib/liblldbPluginObjectContainerMachOFileset.a  lib/liblldbPluginObjectFileBreakpad.a  lib/liblldbPluginObjectFileCOFF.a  lib/liblldbPluginObjectFileELF.a  lib/liblldbPluginObjectFileJSON.a  lib/liblldbPluginObjectFileMachO.a  lib/liblldbPluginObjectFileMinidump.a  lib/liblldbPluginObjectFilePDB.a  lib/liblldbPluginObjectFilePECOFF.a  lib/liblldbPluginObjectFilePlaceholder.a  lib/liblldbPluginObjectFileWasm.a  lib/liblldbPluginPlatformAndroid.a  lib/liblldbPluginPlatformFreeBSD.a  lib/liblldbPluginPlatformGDB.a  lib/liblldbPluginPlatformLinux.a  lib/liblldbPluginPlatformMacOSX.a  lib/liblldbPluginPlatformNetBSD.a  lib/liblldbPluginPlatformOpenBSD.a  lib/liblldbPluginPlatformQemuUser.a  lib/liblldbPluginPlatformWindows.a  lib/liblldbPluginScriptedProcess.a  lib/liblldbPluginProcessGDBRemote.a  lib/liblldbPluginProcessElfCore.a  lib/liblldbPluginProcessMachCore.a  lib/liblldbPluginProcessMinidump.a  lib/liblldbPluginClangREPL.a  lib/liblldbPluginRegisterTypeBuilderClang.a  lib/liblldbPluginScriptInterpreterNone.a  lib/liblldbPluginStructuredDataDarwinLog.a  lib/liblldbPluginSymbolFileBreakpad.a  lib/liblldbPluginSymbolFileCTF.a  lib/liblldbPluginSymbolFileDWARF.a  lib/liblldbPluginSymbolFileJSON.a  lib/liblldbPluginSymbolFilePDB.a  lib/liblldbPluginSymbolFileSymtab.a  lib/liblldbPluginSystemRuntimeMacOSX.a  lib/liblldbPluginSymbolVendorELF.a  lib/liblldbPluginSymbolVendorPECOFF.a  lib/liblldbPluginSymbolVendorWasm.a  lib/liblldbPluginTraceExporterCTF.a  lib/liblldbPluginTypeSystemClang.a  lib/liblldbPluginUnwindAssemblyInstEmulation.a  lib/liblldbPluginUnwindAssemblyX86.a  lib/liblldbPluginProcessPOSIX.a  lib/liblldbPluginObjectContainerMachOFileset.a  lib/liblldbPluginPlatformPOSIX.a  lib/liblldbPluginPlatformGDB.a  lib/liblldbPluginProcessGDBRemote.a  /nix/store/lprrvg44cyn9ajpq77lh7rlnp4kzdqzs-zlib-1.3/lib/libz.so  lib/liblldbPluginDynamicLoaderDarwinKernel.a  lib/liblldbPluginDynamicLoaderMacOSXDYLD.a  lib/liblldbPluginObjectFileMachO.a  lib/liblldbPluginObjectFilePlaceholder.a  lib/liblldbPluginDynamicLoaderPosixDYLD.a  lib/liblldbPluginProcessElfCore.a  lib/liblldbPluginDynamicLoaderPosixDYLD.a  lib/liblldbPluginProcessElfCore.a  lib/liblldbPluginObjectFileBreakpad.a  lib/liblldbPluginObjectFileELF.a  lib/liblldbPluginObjectFilePECOFF.a  lib/liblldbPluginObjectFileWasm.a  lib/liblldbPluginTraceExporterCommon.a  lib/liblldbBreakpoint.a  lib/liblldbCore.a  lib/liblldbDataFormatters.a  lib/liblldbExpression.a  lib/liblldbInterpreter.a  lib/liblldbSymbol.a  lib/liblldbTarget.a  lib/liblldbPluginCPlusPlusLanguage.a  lib/liblldbPluginObjCLanguage.a  lib/liblldbPluginAppleObjCRuntime.a  lib/liblldbPluginObjectFilePDB.a  lib/liblldbPluginSymbolFileDWARF.a  lib/liblldbPluginSymbolFilePDB.a  lib/liblldbPluginTypeSystemClang.a  lib/liblldbCommands.a  lib/liblldbPluginProcessUtility.a  lib/liblldbPluginClangCommon.a  lib/liblldbPluginCPPRuntime.a  lib/liblldbPluginExpressionParserClang.a  lib/liblldbPluginSymbolFileNativePDB.a  lib/liblldbPluginObjCRuntime.a  lib/liblldbBreakpoint.a  lib/liblldbCore.a  lib/liblldbDataFormatters.a  lib/liblldbExpression.a  lib/liblldbInterpreter.a  lib/liblldbSymbol.a  lib/liblldbTarget.a  lib/liblldbPluginCPlusPlusLanguage.a  lib/liblldbPluginObjCLanguage.a  lib/liblldbPluginAppleObjCRuntime.a  lib/liblldbPluginObjectFilePDB.a  lib/liblldbPluginSymbolFileDWARF.a  lib/liblldbPluginSymbolFilePDB.a  lib/liblldbPluginTypeSystemClang.a  lib/liblldbCommands.a  lib/liblldbPluginProcessUtility.a  lib/liblldbPluginClangCommon.a  lib/liblldbPluginCPPRuntime.a  lib/liblldbPluginExpressionParserClang.a  lib/liblldbPluginSymbolFileNativePDB.a  lib/liblldbPluginObjCRuntime.a  lib/liblldbBreakpoint.a  lib/liblldbCore.a  lib/liblldbDataFormatters.a  lib/liblldbExpression.a  lib/liblldbInterpreter.a  lib/liblldbSymbol.a  lib/liblldbTarget.a  lib/liblldbPluginCPlusPlusLanguage.a  lib/liblldbPluginObjCLanguage.a  lib/liblldbPluginAppleObjCRuntime.a  lib/liblldbPluginObjectFilePDB.a  lib/liblldbPluginSymbolFileDWARF.a  lib/liblldbPluginSymbolFilePDB.a  lib/liblldbPluginTypeSystemClang.a  lib/liblldbCommands.a  lib/liblldbPluginProcessUtility.a  lib/liblldbPluginClangCommon.a  lib/liblldbPluginCPPRuntime.a  lib/liblldbPluginExpressionParserClang.a  lib/liblldbPluginSymbolFileNativePDB.a  lib/liblldbPluginObjCRuntime.a  lib/liblldbBreakpoint.a  lib/liblldbCore.a  lib/liblldbDataFormatters.a  lib/liblldbExpression.a  lib/liblldbInterpreter.a  lib/liblldbSymbol.a  lib/liblldbTarget.a  lib/liblldbPluginCPlusPlusLanguage.a  lib/liblldbPluginObjCLanguage.a  lib/liblldbPluginAppleObjCRuntime.a  lib/liblldbPluginObjectFilePDB.a  lib/liblldbPluginSymbolFileDWARF.a  lib/liblldbPluginSymbolFilePDB.a  lib/liblldbPluginTypeSystemClang.a  lib/liblldbCommands.a  lib/liblldbPluginProcessUtility.a  lib/liblldbPluginClangCommon.a  lib/liblldbPluginCPPRuntime.a  lib/liblldbPluginExpressionParserClang.a  lib/liblldbPluginSymbolFileNativePDB.a  lib/liblldbPluginObjCRuntime.a  lib/liblldbBreakpoint.a  lib/liblldbCore.a  lib/liblldbDataFormatters.a  lib/liblldbExpression.a  lib/liblldbInterpreter.a  lib/liblldbSymbol.a  lib/liblldbTarget.a  lib/liblldbPluginCPlusPlusLanguage.a  lib/liblldbPluginObjCLanguage.a  lib/liblldbPluginAppleObjCRuntime.a  lib/liblldbPluginObjectFilePDB.a  lib/liblldbPluginSymbolFileDWARF.a  lib/liblldbPluginSymbolFilePDB.a  lib/liblldbPluginTypeSystemClang.a  lib/liblldbCommands.a  lib/liblldbPluginProcessUtility.a  lib/liblldbPluginClangCommon.a  lib/liblldbPluginCPPRuntime.a  lib/liblldbPluginExpressionParserClang.a  lib/liblldbPluginSymbolFileNativePDB.a  lib/liblldbPluginObjCRuntime.a  /nix/store/fmkdf1b03w5v9yzk2mm3b4yqg1gwhrf5-ncurses-6.4/lib/libpanel.so  /nix/store/fmkdf1b03w5v9yzk2mm3b4yqg1gwhrf5-ncurses-6.4/lib/libncurses.so  /nix/store/fmkdf1b03w5v9yzk2mm3b4yqg1gwhrf5-ncurses-6.4/lib/libform.so  lib/liblldbVersion.a  lib/liblldbHost.a  /nix/store/8g3myidcxw02mv9pzyvwq9cv0y9nayi7-libxml2-2.11.5/lib/libxml2.so  /nix/store/n0w7zjqmks378wvd2r6rbg4za5wv54y2-libedit-20221030-3.1/lib/libedit.so  lib/liblldbUtility.a  -ldl  /nix/store/vbdakqf8mwamgm2s5jfbc54mf2mhljfw-clang-18.0.0-unstable-2023-10-04-lib/lib/libclang-cpp.so.18git  /nix/store/y6hbc39dib6casgmv1lz10s9mia2jqv9-llvm-18.0.0-unstable-2023-10-04-lib/lib/libLLVM-18git.so && :
/nix/store/fzlkaj1ax7gl655blfcr6zzvml1vx3bj-binutils-2.40/bin/ld: cannot open linker script file /build/lldb-src-18.0.0-unstable-2023-10-04/lldb/build/source/API/liblldb.exports: No such file or directory
collect2: error: ld returned 1 exit status
[990/996] Linking CXX executable bin/lldb-server
ninja: build stopped: subcommand failed.
```

--- a/pkgs/development/compilers/llvm/common/lldb.nix
+++ b/pkgs/development/compilers/llvm/common/lldb.nix
@@ -64,7 +64,10 @@ stdenv.mkDerivation (rec {
     lua5_3
   ] ++ lib.optionals enableManpages [
     python3.pkgs.sphinx
+  ] ++ lib.optionals (lib.versionOlder release_version "18" && enableManpages) [
     python3.pkgs.recommonmark
+  ] ++ lib.optionals (lib.versionAtLeast release_version "18" && enableManpages) [
+    python3.pkgs.myst-parser
   ] ++ lib.optionals (lib.versionAtLeast release_version "14") [
     ninja
   ];

--- a/pkgs/development/compilers/llvm/common/lldb.nix
+++ b/pkgs/development/compilers/llvm/common/lldb.nix
@@ -5,7 +5,8 @@
 , cmake
 , zlib
 , ncurses
-, swig
+, swig3
+, swig4
 , which
 , libedit
 , libxml2
@@ -34,6 +35,11 @@ let
         cp -r ${monorepoSrc}/cmake "$out"
         cp -r ${monorepoSrc}/lldb "$out"
       '' else src;
+  vscodeExt = {
+    name = if lib.versionAtLeast release_version "18" then "lldb-dap" else "lldb-vscode";
+    version = if lib.versionAtLeast release_version "18" then "0.2.0" else "0.1.0";
+  };
+  swig = if lib.versionAtLeast release_version "18" then swig4 else swig3;
 in
 
 stdenv.mkDerivation (rec {
@@ -154,14 +160,14 @@ stdenv.mkDerivation (rec {
 
     # Editor support
     # vscode:
-    install -D ../tools/lldb-vscode/package.json $out/share/vscode/extensions/llvm-org.lldb-vscode-0.1.0/package.json
-    mkdir -p $out/share/vscode/extensions/llvm-org.lldb-vscode-0.1.0/bin
-    ln -s $out/bin/*-vscode $out/share/vscode/extensions/llvm-org.lldb-vscode-0.1.0/bin
+    install -D ../tools/${vscodeExt.name}/package.json $out/share/vscode/extensions/llvm-org.${vscodeExt.name}-${vscodeExt.version}/package.json
+    mkdir -p $out/share/vscode/extensions/llvm-org.${vscodeExt.name}-${vscodeExt.version}/bin
+    ln -s $out/bin/*${if lib.versionAtLeast release_version "18" then vscodeExt.name else "-vscode"} $out/share/vscode/extensions/llvm-org.${vscodeExt.name}-${vscodeExt.version}/bin
   '';
 
-  passthru.vscodeExtName = "lldb-vscode";
+  passthru.vscodeExtName = vscodeExt.name;
   passthru.vscodeExtPublisher = "llvm";
-  passthru.vscodeExtUniqueId = "llvm-org.lldb-vscode-0.1.0";
+  passthru.vscodeExtUniqueId = "llvm-org.${vscodeExt.name}-${vscodeExt.version}";
 
   meta = llvm_meta // {
     homepage = "https://lldb.llvm.org/";

--- a/pkgs/development/compilers/llvm/git/clang/default.nix
+++ b/pkgs/development/compilers/llvm/git/clang/default.nix
@@ -21,6 +21,7 @@ let
     sourceRoot = "${src.name}/${pname}";
 
     nativeBuildInputs = [ cmake ninja python3 ]
+      ++ lib.optional (lib.versionAtLeast version "18" && enableManpages) python3.pkgs.myst-parser
       ++ lib.optional enableManpages python3.pkgs.sphinx
       ++ lib.optional stdenv.hostPlatform.isDarwin fixDarwinDylibNames;
 

--- a/pkgs/development/compilers/llvm/git/clang/default.nix
+++ b/pkgs/development/compilers/llvm/git/clang/default.nix
@@ -70,7 +70,7 @@ let
       ln -sv $out/bin/clang $out/bin/cpp
 
       mkdir -p $lib/lib/clang
-      mv $lib/lib/18 $lib/lib/clang/18
+      mv $lib/lib/${lib.versions.major version} $lib/lib/clang/${lib.versions.major version}
 
       # Move libclang to 'lib' output
       moveToOutput "lib/libclang.*" "$lib"

--- a/pkgs/development/compilers/llvm/git/clang/purity.patch
+++ b/pkgs/development/compilers/llvm/git/clang/purity.patch
@@ -4,26 +4,22 @@ Date: Thu, 18 May 2017 11:56:12 -0500
 Subject: [PATCH] "purity" patch for 5.0
 
 ---
- lib/Driver/ToolChains/Gnu.cpp | 7 -------
- 1 file changed, 7 deletions(-)
+ lib/Driver/ToolChains/Gnu.cpp | 3 ---
+ 1 file changed, 3 deletions(-)
 
 diff --git a/lib/Driver/ToolChains/Gnu.cpp b/lib/Driver/ToolChains/Gnu.cpp
 index fe3c0191bb..c6a482bece 100644
 --- a/lib/Driver/ToolChains/Gnu.cpp
 +++ b/lib/Driver/ToolChains/Gnu.cpp
-@@ -487,13 +487,7 @@ void tools::gnutools::Linker::ConstructJob(Compilation &C, const JobAction &JA,
-   } else {
-     if (Args.hasArg(options::OPT_rdynamic))
-       CmdArgs.push_back("-export-dynamic");
-
--    if (!Args.hasArg(options::OPT_shared) && !IsStaticPIE &&
--        !Args.hasArg(options::OPT_r)) {
+@@ -446,9 +446,6 @@ void tools::gnutools::Linker::ConstructJob(Compilation &C, const JobAction &JA,
+                            ToolChain.isPIEDefault(Args));
+       if (IsPIE)
+         CmdArgs.push_back("-pie");
 -      CmdArgs.push_back("-dynamic-linker");
 -      CmdArgs.push_back(Args.MakeArgString(Twine(D.DyldPrefix) +
 -                                           ToolChain.getDynamicLinker(Args)));
--    }
+     }
    }
  
-   CmdArgs.push_back("-o");
 -- 
 2.11.0

--- a/pkgs/development/compilers/llvm/git/default.nix
+++ b/pkgs/development/compilers/llvm/git/default.nix
@@ -19,9 +19,9 @@
 # LLVM release information; specify one of these but not both:
 , gitRelease ? {
   version = "18.0.0";
-  rev = "6f44f87011cd52367626cac111ddbb2d25784b90";
-  rev-version = "18.0.0-unstable-2023-10-04";
-  sha256 = "sha256-CqsCDlzg8I2c9BybKP7B5nfHiQWktqgVavrfiYkjkx4=";
+  rev = "2fd7657b6609454af7adb75765d164ec7d1bb80b";
+  rev-version = "18.0.0-unstable-2023-12-13";
+  sha256 = "sha256-/sMQzzFid0tAnreOIV9SUm2H6QbEGhpNcizl3LDPM5s=";
 }
   # i.e.:
   # {

--- a/pkgs/development/compilers/llvm/git/llvm/default.nix
+++ b/pkgs/development/compilers/llvm/git/llvm/default.nix
@@ -86,10 +86,14 @@ stdenv.mkDerivation (rec {
 
   nativeBuildInputs = [ cmake ninja python ]
     ++ optionals enableManpages [
-      # Note: we intentionally use `python3Packages` instead of `python3.pkgs`;
-      # splicing does *not* work with the latter. (TODO: fix)
-      python3Packages.sphinx python3Packages.recommonmark
-    ];
+    # Note: we intentionally use `python3Packages` instead of `python3.pkgs`;
+    # splicing does *not* work with the latter. (TODO: fix)
+    python3Packages.sphinx
+  ] ++ optionals (lib.versionOlder version "18" && enableManpages) [
+    python3Packages.recommonmark
+  ] ++ optionals (lib.versionAtLeast version "18" && enableManpages) [
+    python3Packages.myst-parser
+  ];
 
   buildInputs = [ libxml2 libffi ]
     ++ optional enablePFM libpfm; # exegesis

